docs/examples/generate_diffs.sh

```diff
@@ -24,19 +24,19 @@ generate_diff() {
 pushd "${_SCRIPT_DIR}"
 
 # single_gpu -> multi_gpu
-generate_diff distributed/001_single_gpu/job.sh distributed/002_multi_gpu/job.sh
-generate_diff distributed/001_single_gpu/main.py distributed/002_multi_gpu/main.py
+generate_diff distributed/single_gpu/job.sh distributed/multi_gpu/job.sh
+generate_diff distributed/single_gpu/main.py distributed/multi_gpu/main.py
 
 # multi_gpu -> multi_node
-generate_diff distributed/002_multi_gpu/job.sh distributed/003_multi_node/job.sh
-generate_diff distributed/002_multi_gpu/main.py distributed/003_multi_node/main.py
+generate_diff distributed/multi_gpu/job.sh distributed/multi_node/job.sh
+generate_diff distributed/multi_gpu/main.py distributed/multi_node/main.py
 
 # single_gpu -> checkpointing
-generate_diff distributed/001_single_gpu/job.sh good_practices/checkpointing/job.sh
-generate_diff distributed/001_single_gpu/main.py good_practices/checkpointing/main.py
+generate_diff distributed/single_gpu/job.sh good_practices/checkpointing/job.sh
+generate_diff distributed/single_gpu/main.py good_practices/checkpointing/main.py
 
 # single_gpu -> hpo_with_orion
-generate_diff distributed/001_single_gpu/job.sh good_practices/hpo_with_orion/job.sh
-generate_diff distributed/001_single_gpu/main.py good_practices/hpo_with_orion/main.py
+generate_diff distributed/single_gpu/job.sh good_practices/hpo_with_orion/job.sh
+generate_diff distributed/single_gpu/main.py good_practices/hpo_with_orion/main.py
```

Comment:
Could we loop over `**/job.sh` and `**/main.py` instead, so we don't have to remember to update this file whenever we add a new example?

Reply:
Perhaps I don't understand, but here we need to define the from-to relationship between examples. How do you suggest we do that automatically?

Reply:
Ah yeah I had a brain fart here, sorry.

Although, one way to do this automatically would be to represent the examples as a Git arborescence: you write the code for single_gpu, then you create a multi_gpu branch that has one commit that is the difference, and so on, each example being its own branch. Then you can run a series of checkouts and cp to populate the examples directory in the actual docs repo. With this method you can also update the whole set of examples more easily by changing the base and looping on git rebase. But in any case this is out of scope for this PR.

Reply:
Yeah that's interesting!

I also played around with the idea of us editing the .diff files directly instead of the example files, and re-generating the derived examples if something changes in the base.

For now, with this limited number example dependencies, it's pretty manageable to make a change in the base example and then copy-paste it to the 5-6 derived examples. However there is potentially a better solution in the long run for sure.
I agree with you, I don't think it's worth it to spend too much time working on that right now.